src/main.rs

```diff
@@ -31,6 +42,27 @@ struct UsageError {
     message: String,
 }
 
+/// Parses the value of a flag that takes a duration.
+///
+/// At the moment, the only supported unit for the duration are seconds and the textual value
+/// must include the "ms" suffix to support adding extra units in the future if desired.
+fn parse_duration(value: &str) -> Result<Timespec, UsageError> {
```

Comment:
This implementation of parser is harder to extend to new units and makes it difficult to give good error messages. Why not do something like the following to define the protocol between the value and units so that you can say things like "invalid unit 'ms'"?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=34710115ee2ea47323746c1bf8d1f011